test/scripts/syscall_invocation_test.py

```diff
@@ -0,0 +1,574 @@
+            # };
+            p32 = lambda x: int.to_bytes(x, 4, "little")
+    
+            struct_bytes = \
+                p32(MAP_BASE_1) + \
+                p32(0x1000) + \
+                p32(prot) + \
+                p32(flags) + \
+                p32(0xffffffff) + \
+                p32(0)
+            
+            # We are goint to write the struct to a stack address
+            stack = d.maps.filter("stack")[0]
+            d.memory[stack.start, len(struct_bytes), "absolute"] = struct_bytes
+
+            # Invoke the syscall
+            ret = d.invoke_syscall("mmap", stack.start)
+        else:
+            # unsigned long addr, unsigned long len, unsigned long prot, unsigned long flags, unsigned long fd, unsigned long off
+            ret = d.invoke_syscall("mmap", MAP_BASE_1, 0x1000, prot, flags, -1, 0)
+
+        # Page aligned address should be returned
+        self.assertEqual(ret, MAP_BASE_1)
+
+        post_num_maps = len(d.maps)
+        self.assertGreater(post_num_maps, prev_num_maps)
+
+        # Check protection
+        mmap_map = d.maps.filter(ret)[0]
+        self.assertEqual(mmap_map.permissions, "rwxp")
+
+        d.terminate()
+    
+    @skipIf(PLATFORM == "aarch64", "Fork not supported on aarch64")
+    def test_fork(self):
+        d = debugger(RESOLVE_EXE("dummy_binary"), aslr=False)
+        d.run()
+        
+        # Set a breakpoint to <main>
+        d.breakpoint(BP_ADDRESS, hardware=True, file="binary")
+
+        d.cont()
+        d.wait()
+
+        ip = d.instruction_pointer
+
+        # Invoke the syscall
+        ret = d.invoke_syscall("fork")
+
+        self.assertEqual(d.instruction_pointer, ip)
+
+        # Check the return value
+        self.assertGreater(ret, 0)
+
+        # Check that the child process is registered
+        self.assertIsNotNone(d.children)
+
+        d.children[0].terminate()
+        d.terminate()
+
+    def test_clone_process(self):
+        d = debugger(RESOLVE_EXE("dummy_binary"), aslr=False)
+        d.run()
+        
+        # Set a breakpoint to <main>
+        d.breakpoint(BP_ADDRESS, hardware=True, file="binary")
+
+        d.cont()
+        d.wait()
+
+        ip = d.instruction_pointer
+
+        # Invoke the syscall
+        SIGCHLD = 17
+
+        # unsigned long clone_flags, unsigned long newsp, int *parent_tidptr, unsigned long tls, int *child_tidptr
+        clone_flags = SIGCHLD
+        stack_base = d.maps.filter("stack")[0].start
+
+        if PLATFORM == "amd64":
+            ret = d.invoke_syscall("clone", clone_flags, stack_base, stack_base + 0x08, stack_base + 0x10, d.regs.fs_base)
+        # For some obscure reason, the last two arguments are swapped in many architectures
+        elif PLATFORM == "i386":
+            ret = d.invoke_syscall("clone", clone_flags, stack_base, stack_base + 0x04, d.regs.gs, stack_base + 0x08)
+        elif PLATFORM == "aarch64":
+            # To retrieve the TLS base, we need to use the TPIDR_EL0 register
```

Comment:
The `tls` parameter is nullable in the clone syscall.
You are not setting the CLONE_SETTLS flag, so I don't think that this is needed, just pass 0x0 and it will work.
I think this is the same for the other two parameters as well, `parent_tid` and `child_tid`.